### PR TITLE
Mixpanel Integration

### PR DESCRIPTION
**Pull request by thegoldenmule**
- Added simple tracking API.
- Added MixPanel tracking implementation.
- Included mixpanel debug and release tokens.

### CODE REVIEW

File: Assets/Source/Util/UserTracking/MixPanelTrackingService.cs [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+                }
+
+                payload["$mp_metadata"] = GenerateEventMetadata();
+
+                // send
+                _bootstrapper.BootstrapCoroutine(Send(payload));
+            }
+            catch (Exception ex)
+            {
+                Log.Error(this, "Woe is me : {0}.", ex);
```
Comment (gdamron) [inline Assets/Source/Util/UserTracking/MixPanelTrackingService.cs]: Ha, maybe a more descriptive message :)

File: Assets/Source/Util/UserTracking/TrackingEventTypes.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+﻿namespace CreateAR.EnkluPlayer.UserTracking
+{
+    /// <summary>
+    /// Describes all the potential event types to track.
+    /// </summary>
+    public static class TrackingEventTypes
+    {
+        public const string SIGNOUT = "signout";
```
Comment (gdamron) [inline Assets/Source/Util/UserTracking/TrackingEventTypes.cs]: This doesn't appear to be used. Did you mean to create an event in `SignOutApplicationState`?

**Comment by jsundra** [comment]
How do you feel about adding a method to `ITrackingService` to register/add new globals? I could see it being useful for something like the currently running appId to be registered as a global so that as we add more calls we can correlate better without having to always have the appId handy when sending an event.

Then of course `Reset` would unregister/clear the non-internal globals that the tracking service was given.

File: Assets/Source/Util/UserTracking/MixPanelTrackingService.cs (hunk 1 above)
Comment (thegoldenmule) [reply Assets/Source/Util/UserTracking/MixPanelTrackingService.cs]: Whoops! I'll remove this.

File: Assets/Source/Util/UserTracking/TrackingEventTypes.cs (hunk 2 above)
Comment (thegoldenmule) [reply Assets/Source/Util/UserTracking/TrackingEventTypes.cs]: Why... yes I did!

**Comment by thegoldenmule** [comment]
@jsundra The de facto mixpanel SDK has this notion, it's a good idea. I will leave it as an exercise for the next user-- when it's needed. :)

File: Assets/Source/Util/UserTracking/MixPanelTrackingService.cs [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+        /// <inheritdoc />
+        public void Identify(string userId)
+        {
+            if (DistinctId == userId)
+            {
+                return;
+            }
+
+            var previousId = DistinctId;
+
```
[the remark this answers is not in the document]
Comment (thegoldenmule) [reply Assets/Source/Util/UserTracking/MixPanelTrackingService.cs]: Blast! No, it is supposed to be stored.

**Pull request merged**